### PR TITLE
adding service HealthCheckInterval implementation

### PR DESCRIPTION
**Pull request by jeremymv2**
This is required for https://github.com/habitat-sh/habitat/pull/5890

@cm @baumanj have already initially reviewed the above PR and therefore will have some context on this change.

Signed-off-by: Jeremy J. Miller <jm@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/core/src/service.rs [hunk 1]
```diff
@@ -350,11 +357,53 @@ impl FromStr for ApplicationEnvironment {
     }
 }
 
+/// Represents how far apart to run health checks for individual services
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct HealthCheckInterval(pub Duration);
```
[the remark this answers is not in the document]
Comment (baumanj) [reply components/core/src/service.rs]: Another alternative that I think I may prefer is to get rid of the nanosecond precision (we really don't need it for this type) and just wrap a u64:
```rust
pub struct HealthCheckInterval {
    seconds: u64,
}
```
This means we don't implement `AsRef` and instead convert to `Duration` as needed. The changes are pretty straightforward and I think this is the best of both worlds.

File: components/core/src/service.rs [hunk 2]
```diff
@@ -350,11 +357,53 @@ impl FromStr for ApplicationEnvironment {
...
+    fn from(d: Duration) -> Self {
+        HealthCheckInterval(d)
+    }
+}
+
+impl Into<Duration> for HealthCheckInterval {
+    fn into(self) -> Duration {
+        self.0
+    }
+}
```
Comment (baumanj) [inline components/core/src/service.rs]: [Per the docs](url), implementing `From` is preferred:
> Library authors should not directly implement this trait, but should prefer implementing the From trait, which offers greater flexibility and provides an equivalent Into implementation for free, thanks to a blanket implementation in the standard library.

So, something like:
```rust
impl From<HealthCheckInterval> for Duration {
    fn from(h: HealthCheckInterval) -> Self {
        …
    }
}
```

**Comment by jeremymv2** [comment]
@baumanj @christophermaier Some recent spare time over the weekend allowed me to come back to this and it is now ready for a re-review at your convenience.

**Comment by jeremymv2** [comment]
@baumanj see https://github.com/habitat-sh/core/pull/95/commits/7874075ed63d566bf5224b399d7c79ec1f9fc6c0 😄

**Comment by baumanj** [comment]
I'll merge this once our current release process is finished.

**Pull request merged**